tools/rpm2img

```diff
@@ -227,15 +227,19 @@ set default="0"
 set timeout="0"
 
 menuentry "${PRETTY_NAME} ${VERSION_ID}" {
-   linux (\$root)/vmlinuz root=/dev/dm-0 \\
+   linux (\$root)/vmlinuz \\
        ${KERNEL_PARAMETERS} \\
-       rootwait ro \\
+       bootconfig \\
+       root=/dev/dm-0 rootwait ro \\
        raid=noautodetect \\
        random.trust_cpu=on selinux=1 enforcing=1 \\
-       systemd.log_target=journal-or-kmsg systemd.log_color=0 net.ifnames=0 \\
-       biosdevname=0 dm_verity.max_bios=-1 dm_verity.dev_wait=1 \\
+       dm_verity.max_bios=-1 dm_verity.dev_wait=1 \\
        dm-mod.create="root,,,ro,0 $VERITY_DATA_512B_BLOCKS verity $VERITY_VERSION PARTUUID=\$boot_uuid/PARTNROFF=1 PARTUUID=\$boot_uuid/PARTNROFF=2 \\
-       $VERITY_DATA_BLOCK_SIZE $VERITY_HASH_BLOCK_SIZE $VERITY_DATA_4K_BLOCKS 1 $VERITY_HASH_ALGORITHM $VERITY_ROOT_HASH $VERITY_SALT 1 restart_on_corruption"
+       $VERITY_DATA_BLOCK_SIZE $VERITY_HASH_BLOCK_SIZE $VERITY_DATA_4K_BLOCKS 1 $VERITY_HASH_ALGORITHM $VERITY_ROOT_HASH $VERITY_SALT 1 restart_on_corruption" \\
+       -- \\
+       systemd.log_target=journal-or-kmsg systemd.log_color=0 \\
+       net.ifnames=0 biosdevname=0
+   initrd (\$private)/bootconfig.data
```

Comment:
If we can, it might be better to create an empty but valid bootconfig file in order to prevent the kernel warning from being printed.

Reply:
I think an empty file will still cause the warning to be printed since it will find the file, but no bootconfig.  I think I tested that, but I'll double check and follow up.

Reply:
@bcressey: I confirmed that an empty bootconfig file still logs the message since no bootconfig data exists in the file
```
bash-5.0# cat /var/lib/bottlerocket/bootconfig.data 
bash-5.0#

bash-5.0# journalctl | grep -i bootconfig
...
Mar 15 21:27:57 localhost kernel: 'bootconfig' found on command line, but no bootconfig found
...
```

Reply:
Can you also make sure that the use of the `$private` variable doesn't cause problems if it's unset?

It wouldn't be in the case of an in-place upgrade on existing variants, since the contents of the BIOS boot partition or EFI system partition are never updated.

Reply:
Good catch - if `$private` is  unset, grub spits out the following error.
```
error: ../../grub-core/net/net.c:1394:no server is specified
```
I haven't dug into any grub code yet so I'm not 100% sure why grub would  diverge into the `net` module because of an unset variable.

The boot does continue after a few moments, but I don't think the delay and error message is acceptable, not to mention the unknown behavior happening behind the scenes.

I've experimented in with excluding the `initrd` line in `rpm2img`.  However, we don't include the `test` module in our build of grub so any `if` statements don't really work.

I'll keep digging.